src/document.jl

```diff
@@ -107,20 +107,20 @@ end
 #### output
 
 function save_file(xdoc::XMLDocument, filename::AbstractString; encoding::AbstractString="utf-8")
-    ret = ccall((:xmlSaveFormatFileEnc,libxml2), Cint, (Cstring, Xptr, Cstring, Cint),
-        filename, xdoc.ptr, encoding, 1)
-    if ret < 0
-        throw(XMLWriteError("Failed to save XML to file $filename"))
-    end
-    return @compat Int(ret)  # number of bytes written
+    ret = ccall((:xmlSaveFormatFileEnc,libxml2), Cint,
+                (Cstring, Xptr, Cstring, Cint),
+                filename, xdoc.ptr, encoding, 1)
+    ret < 0 && throw(XMLWriteError("Failed to save XML to file $filename"))
+    Int(ret)  # number of bytes written
 end
 
-function Base.string(xdoc::XMLDocument; encoding::AbstractString="utf-8")
-    buf_out = Vector{Xstr}(1)
-    len_out = Vector{Cint}(1)
-    ccall((:xmlDocDumpFormatMemoryEnc,libxml2), Void, (Xptr, Ptr{Xstr}, Ptr{Cint}, Cstring, Cint),
-        xdoc.ptr, buf_out, len_out, encoding, 1)
+function string(xdoc::XMLDocument; encoding::AbstractString="utf-8")
+    buf_out = create_vector(Xstr, 1)
+    len_out = create_vector(Cint, 1)
```

Comment:
Would be cleaner to just use `Compat` and `Vector{Cint}(uinitialized, 1)`

Reply:
Force of habit - I tend to avoid bringing such a large dependency for my own work, but I'll change this.